cpp/ql/src/jsf/4.10 Classes/AV Rule 85.ql

```diff
@@ -28,14 +28,34 @@ predicate implementedAsNegationOf(Operator op1, Operator op2) {
     c.getTarget() = op2)
 }
 
+predicate classIsCheckableFor(Class c, string op) {
+  oppositeOperators(op, _) and
+  if exists(Class templ | c.isConstructedFrom(templ))
+  then // If we are an instantiation, then we can't check `op` if the
+       // template has it but we don't (because that member wasn't
+       // instantiated)
+       exists(Class templ | c.isConstructedFrom(templ) and
+                            (templ.getAMember().hasName(op) implies
+                             exists(Function f | f = c.getAMember() and
+                                                 f.hasName(op) and
+                                                 f.isDefined())))
+  else any()
```

Comment:
Would this be more readable without the `if`, as `exists(Class temp1 | c.isConstructedFrom(temp1) and ...) or not exists(Class temp1 | c.isConstructedFrom(temp1))`?

Reply:
I changed it to be an `implies`